### DISCUSSION TITLE
🚀 Distro 2023.05.0 Released

### DISCUSSION THREAD

**Discussion by Nicogene** [robotology/community#639] [Releases]
# 2023.05.0 Distro Announcement

## Brief description of the announcement

Hello @robotology/everyone  👋

The [iCub project](https://icub.iit.it/) is happy to announce our **Software Distro 2023.05.0** 📦

This is a major release that comprises **binary packages** as well. Therefore, follow the [instructions][1] available in our documentation to upgrade your system and/or download the new dependencies.

You can find the tags composing this distro in our [Software Versioning Table][2].

## Detailed context

This distro is shipped with several updates and fixes, including the ones listed hereafter:

#### 🔘 `embObjFTSensor` replaces `embObjStrain`

`embObjFTSensor` now handles also `strain` board, this made `embObjStrain` device obsolete, then it has been removed.

- https://github.com/robotology/icub-main/issues/873
- https://github.com/robotology/icub-main/issues/875

#### 🔘 Refactored how FT sensors data is published

This Distro ship a deep refactor on how the FT sensors data is published both on real and simulated robots. Now both simulated and real robot publish their Force-Torque Sensors measurements **only** via the [`multipleanalogsensorsserver`](https://www.yarp.it/latest//classMultipleAnalogSensorsServer.html) YARP device. The `name`  parameter passed to this device is: `/<robotPortPrefix>/<partName>/FT`.  This means that for each part there will be a `multipleanalogsensorsserver` device that will open the following YARP ports:
  * `/<robotPortPrefix>/<partName>/FT/measures:o` : that publishes sensors information for the part, using the structure defined in https://github.com/robotology/yarp/blob/master/src/devices/multipleAnalogSensorsMsgs/multipleAnalogSensorsSerializations.thrift
  * `/<robotPortPrefix>/<partName>/FT/rpc:o` : that expose several information related to the part via a YARP RPC port

These YARP ports are not meant to be accessed directly, but should be accessed instead via the [`multipleanalogsensorsclient`](https://www.yarp.it/git-master/classMultipleAnalogSensorsClient.html) device.

The publishing via `AnalogServer` has been removed since this Distro.
Here some development:
- https://github.com/robotology/robots-configuration/issues/470
- https://github.com/robotology/robots-configuration/issues/485
- https://github.com/robotology/icub-models-generator/pull/239
- https://github.com/icub-tech-iit/ergocub-software/pull/123

#### 🔘 Fixed launching face expressions on iCub
We fixed the face expressions, which now rely on the use of `serialPort_nws_yarp` through the following PR's:
- https://github.com/robotology/icub-main/pull/883
- https://github.com/robotology/icub-main/pull/877
- https://github.com/robotology/icub-main/pull/888

#### 🔘 Introduced `robots-configuration` CI dry-run
This is the first distro sporting the **CI dry-run of the robot configuration files**.

The goal is to check the consistency of XML files (e.g., missing files, malformed syntax...) as well as of the device drivers launched with the robot (e.g., missing parameters, deprecated syntax...) within the [CI of the `robots-configuration` repository][ci-dry-run] for each proposed changes through PR's.
This way, we can easily detect errors and leftovers ahead of the deployment.

The outcome of the dry-run is returned as an artifact within the CI workflow.

[ci-dry-run]: https://github.com/robotology/robots-configuration/blob/master/.github/workflows/ci.yml

#### 🔘 Calibration 14 
This Distro provides the new calibration 14: it has been developed for calibrating the joint with the absolute encoder at the joint (which currently works with POS service / FAP encoder sensor) and with incremental encoder at the motor. The calibration 14 is useful to calibrate the incremental encoder reaching a hardware stop and thus ensure that the movement of the rotor is always within the limits specified by the user, avoiding damage to the motor itself.

#### 🔘 Deb binaries

As discussed [here](https://github.com/orgs/robotology/discussions/612), we decided to deprecate the apt binaries in favor of conda ones that are delivered for both Linux, Windows and macOS.
Please refer to the [official documentation](https://icub-tech-iit.github.io/documentation/sw_installation/conda_binaries/) for further details.

[1]: https://icub-tech-iit.github.io/documentation/sw_installation/
[2]: https://icub-tech-iit.github.io/documentation/sw_versioning_table/#table

[^1]: See https://github.com/robotology/icub-main/pull/808
[^2]: See https://github.com/robotology/icub-main/issues/842#issuecomment-1327388046

**Comment by traversaro** [comment]
If anyone is having problems with the release 2023.05.0 to read FT sensors on real iCub, please note that we discovered a bug related to that that is being fixed by https://github.com/robotology/icub-main/pull/892 .

**Comment by Nicogene** [comment]
Today we released the [Distro 2023.05.2](https://icub-tech-iit.github.io/documentation/sw_versioning_table/2023.05.2) which contains some important fixes/features such as:

- https://github.com/robotology/icub-main/pull/893
- https://github.com/robotology/icub-main/pull/892
- https://github.com/robotology/icub-models-generator/pull/240
- https://github.com/robotology/icub-models-generator/pull/241
- https://github.com/robotology/robots-configuration/pull/535
- https://github.com/robotology/robots-configuration/pull/534
- https://github.com/robotology/robots-configuration/pull/533

**Reply by pattacini** [reply]
Pay attention in particular to the introduction of the [Robots Icebox](https://github.com/robotology/robots-configuration/pull/533) in `robots-configuration`.

More on this at https://github.com/robotology/robots-configuration/blob/master/robots-icebox/README.md.
